compiler/runtime/OMRRuntimeAssumptions.cpp

```diff
@@ -40,3 +40,88 @@ void TR::PatchNOPedGuardSite::compensate(bool isSMP, uint8_t *location, uint8_t
    {
    _patchVirtualGuard(location, destination, isSMP);
    }
+
+TR::PatchSites::PatchSites(TR_PersistentMemory *pm, size_t maxSize) :
+    _maxSize(maxSize),
+    _size(0),
+    _refCount(0),
+    _patchPoints(NULL),
+    _firstLocation(NULL),
+    _lastLocation(NULL)
+   {
+   _patchPoints = new (pm) uint8_t*[maxSize * 2];
+   }
+
+/**
+ * Add a patch site to the collection.
+ */
+void TR::PatchSites::add(uint8_t *location, uint8_t *destination)
+   {
+   TR_ASSERT(_size < _maxSize, "Cannot add more patch sites, max size is %d", _maxSize);
+   _patchPoints[_size] = location;
+   _patchPoints[_size + 1] = destination;
+   _size++;
+
+   if (_firstLocation == NULL || location < _firstLocation)
+      _firstLocation = location;
+   if (_lastLocation == NULL || location > _lastLocation)
+      _lastLocation = location;
+   }
+
+/**
+ * Compare this collection with another.
+ * Will return true if all locations are the same.
+ */
+bool TR::PatchSites::equals(PatchSites *other)
+   {
+   if (other->getSize() != _size)
+      return false;
+
```

Comment:
Would it make sense to add a fast path exit based on comparing the ranges of the two sets of assumptions - if they don't match exactly then the sets won't be equal right? That should cause complete equality checks in a minority of cases would be my thoughts...

Reply:
Thanks, I have added the fast path.